src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -64,7 +79,8 @@ public Http2OutputProducer(Http2Stream stream, Http2StreamContext context, Strea
 
         public void StreamReset()
         {
-            Debug.Assert(_dataWriteProcessingTask.IsCompletedSuccessfully);
+            // Data background task must still be running.
+            Debug.Assert(!_dataWriteProcessingTask.IsCompleted);
```

Comment:
This assert should be checking the _responseCompleteTaskSource  status is completed. Essentially the _responseCompleteTaskSource  should have been completed by then.